controllers/che/backup.go

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package che
+
+import (
+	"context"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	chev1 "github.com/eclipse-che/che-operator/api/v1"
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+)
+
+func isPreviousVersionDeployed(cheCR *chev1.CheCluster) bool {
+	deployedVersion := cheCR.Status.CheVersion
+	currentVersion := deploy.DefaultCheVersion()
```

Comment:
To get a current version (it is better to name `currentVersion` -> `newVersion` and `deployedVersion` -> `currentVersion`)
https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/server/server.go#L352

Reply:
Disagree, because all the logic is happening in a new operator. So, the new Che operator IS current (running the code now). Deployed version of Che is previous in relation to the current operator.

Reply:
Let's detect Che version basing on operator data instead of Che server image tag.